### PR TITLE
[pvr] json-rpc: Fix clientid argument to scan method to be optional

### PR DESCRIPTION
**Pull request by phunkyfish**
## Description

Fixing API issue introduced by https://github.com/xbmc/xbmc/pull/18647 where internal client ID valid was exposed publicly. Instead the field used is optional now.

As noted by @Tolriq: https://github.com/xbmc/xbmc/pull/18647#issuecomment-719781039

## Types of change

- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by phunkyfish** [comment]
@Tolriq is this good now from an API persective?

File: xbmc/interfaces/json-rpc/PVROperations.cpp [hunk 1]
```diff
@@ -282,8 +283,12 @@ JSONRPC_STATUS CPVROperations::Scan(const std::string &method, ITransportLayer *
...
     return FailedToExecute;
 
-  if (CServiceBroker::GetPVRManager().GUIActions()->StartChannelScan(
-      parameterObject["clientid"].asInteger()))
+  int clientId = PVR_INVALID_CLIENT_ID;
+
+  if (parameterObject.isMember("clientid"))
+    clientId = parameterObject["clientid"].asInteger();
+
+  if (CServiceBroker::GetPVRManager().GUIActions()->StartChannelScan(clientId))
```
Comment (ksooo) [inline xbmc/interfaces/json-rpc/PVROperations.cpp:291]: if you would introduce GUIActions::StartChannelScan() without params, you would eliminate the dependency to PVRClient.h

Comment (phunkyfish) [reply xbmc/interfaces/json-rpc/PVROperations.cpp]: Makes perfect sense.

**Review by ksooo — APPROVED** [review]
:-)

**Comment by Tolriq** [comment]
Much better yes thanks.

**Review by DaveTBlake — APPROVED** [review]
Yeap, this is better. The previous PR grew a lot , I somehow missed this aspect.

**Pull request merged**